reroll-initiative/reroll-initiative.js

```diff
@@ -0,0 +1,203 @@
+            }
+        });
+
+        this._loadSettings();
+
+        /* add settings as object instead
+        game.settings.register('reroll-initiative', "rriStatus", {
+            name: "Reroll-Initiative Status",
+            hint: "Enable the rerolling initiative true/false",
+            default: RRI_CONFIG.reroll,
+            //default: "",
+            type: Boolean,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+
+        game.settings.register('reroll-initiative', "actorTypesToReroll", {
+            name: "Actor Types to Reroll",
+            hint: "Specify the actor types to reroll PCs/NPCs/All",
+            default: RRI_CONFIG.actorTypes,
+            //default: "",
+            type: String,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+        */
+    }
+
+    /**
+     * Returns the default module settings
+     */
+    _defaultSettings() {
+        this.settings = RRI_SETTINGS;
+        console.log("Restting reroll-initiative settings to defaults:",RRI_DEFAULT_SETTINGS);
+    }
+
+    /**
+     * Saves current class instance settings back to game settings
+     */
+    _saveSettings () {
+        game.settings.set("reroll-initiative","rriSettings",this.settings);
+    }
+
+    /**
+     * Loads current class instance settings from game settings
+     */
+    _loadSettings (){
+        let settings = game.settings.get("reroll-initiative","rriSettings");
+        this.settings = settings;
+        console.log(this.settings);
+    }
+    
+    /**
+     * Get the current class instance settings (for external use)
+     */
+    get settings() {
+        this._loadSettings();
+        return this.settings;
+    }
+
+    /**
+     * Change the current class instance settings (for external use)
+     * @param {Object} incomingSettings
+     */
+    set settings(incomingSettings) {
+        this.settings = incomingSettings;
+        return this._saveSettings();
+    }
+
+    /**
+     * @name postUpdateCombatHook
+     * @description Hook on combat update and if round in update is greater than previous -- call resetAndReroll
+     */
+    postUpdateCombatHook() {
+        Hooks.on("updateCombat", (combat,update) =>  {
+            this._loadSettings();
+
+            if(this.settings.reroll){
+                
+                if(update.round && combat.previous && update.round > combat.previous.round){
+                    //console.log("Reroll-Initiative: Round incremented - rerolling initiative")
+                    this.resetAndReroll(combat);
+                }
+            }
+            
+        }); 
+    }
+
+    /**
+     * @name resetAndReroll
+     * @param {Combat} combat
+     * @description For the given combat instance, call the resetAll method and the rollAll method
+     */
```

Comment:
not sure if this should be private. technically it could be called from outside the class (all it really does it take a combat instance and run some methods from the Combat class)

Reply:
If you don't have a solid need for being public, I'd leave it as a private method.